source/ChiPSeqReader.py

```diff
@@ -362,6 +362,32 @@ def duplicate_region(self, interval):
         self.chr_data[interval.chr].sort_values(by=["start"], inplace=True)
         assert len(self.chr_data[interval.chr]) - debug == old_length
 
+    def duplicate_region_v2(self, interval):                     # Modifies data according to the relationship
+        st, en = self.get_interval(interval, return_ids=True)    # between the "start" and "end" of each peak and
+        #print(st, en)                                           # the "start" and "end" of the duplication interval
+        old_length = len(self.chr_data[interval.chr])
+        drop_indices = []
+        dup_indices = []
+        for i in range(st, en + 1):
+            if (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("start")] < interval.start) and \
```

Comment:
Does it handle the case when ChIP-seq interval _overlaps_ (not contains) duplication interval? This is actually the most common case when ChIP-seq peak will be destroyed.
Example:
ChIP-seq peak 100-200
Dupliication interval 150-1 000

Reply:
I agree, you are right!